app/models/concerns/foreman_remote_execution/host_managed_extensions.rb

```diff
@@ -0,0 +1,16 @@
+module ForemanRemoteExecution
+  module HostManagedExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      alias_method_chain :params, :remote_execution
+    end
+
+    def params_with_remote_execution
+      params = params_without_remote_execution
+      keys = remote_execution_ssh_keys
+      params['remote_execution_ssh_keys'] = keys unless keys.blank?
+      params
+    end
+  end
+end
```

Comment:
you might need adding `require_dependency 'host_managed_extensions'` to ensure loading order